### PR TITLE
Container manage dir

### PR DESCRIPTION
**Pull request by willemolding**
Depends on #840 

- Adds the ability to (optionally) specify the directory you want the container to use for its dynamic updated config and DNA files in the container-config.toml. Default is ~/.holochain but maybe we should switch to [projectDirs](https://docs.rs/directories/0.8.5/directories/struct.ProjectDirs.html) like @zippy suggested?

- Adds a boolean flag `store=true` to `install_dna_from_file` to specify if the DNA should be moved to the managed dir

- Adds optional `properties` parameter to `install_dna_from_file`. If given and `copy=true` it will merge the properties from the original DNA with the new properties to produce a new DNA with a new hash which is then installed to the managed dir. This could alternatively be moved to its own rpc method (e.g. `dna/create/from_template`) but I think it is quite neat this way.

### CODE REVIEW

**Comment by willemolding** [comment]
It looks like there is something now quite right with how the file paths are handled on windows

**Comment by lucksus** [comment]
@maackle Moving file handling stuff into `container` is actually a good point, especially in the context of the light client, i.e. being able to compile to WASM. Currently we can't compile `container_api` to WASM anyway because it depends on `cas_implementations` which won't compile to WASM because of the filesystem implementation. But that is ok. The light-client CAS implementation is what will have to be implemented in JS for browser local storage anyway. But it would be cool if we could reuse `container_api` for the light-client. 

How can we extract all file handling, including the filesystem storage implementation, to `container`?

I would vote for going ahead with merging this PR (after the tests pass on Windows) and approach this questions separately..

**Comment by lucksus** [comment]
Pulling in @sphinxc0re as we just talked about this a bit in person.

File: container_api/src/container/admin.rs [hunk 1]
```diff
@@ -111,9 +147,27 @@ impl ContainerAdmin for Container {
         Ok(())
     }
 
-    fn add_instance(&mut self, instance: InstanceConfiguration) -> Result<(), HolochainError> {
+    fn add_instance(
```
[the remark this answers is not in the document]
Comment (willemolding) [reply container_api/src/container/admin.rs]: I think that file storage is a sensible default at least until there is a local db store or similar

**Comment by lucksus** [comment]
@maackle, can we merge this?

**Pull request merged**